### PR TITLE
Issue 155/organizing utils

### LINKED ISSUE #155: Organizing utils
**Is your feature request related to a problem? Please describe.**
At the moment there's been many new functionalities being created for PASS. Many new functionalities are stored within utils. We should consider organizing the files into utils subdirectories.

**Describe the solution you'd like**
As stated above, we can consider organizing files into subdirectories.

### PR DESCRIPTION
**Pull request by leekahung**
Updates (May 21, 2023): As stated in the title, I'm reorganizing utils into 4 general sub-directories:

- barcode
- cryptography
- frontend
- network

I'm open to suggestions for all 4 names. The file index.js is still being used to funnel all the exports from the sub-directories to the rest of src.

The full utils directory will look like the following:

```
└── utils
    ├── barcode
    │   ├── barcode-date-parser.js
    │   └── barcode-scan.js
    ├── cryptography
    │   ├── credentials-helper.js
    │   └── credentials-helper.test.js
    ├── frontend
    │   ├── FormSubmissionHelper.js
    │   ├── FormSubmissionHelper.test.js
    │   ├── form-helper.js
    │   ├── logout-helper.js
    │   └── notification-helper.js
    ├── index.js
    └── network
        ├── session-core.js
        ├── session-helper.js
        └── session-helper.test.js
```

When we decide to refactor session-core, we can decide then where to place within these 4 categories or create a new one altogether.

### CODE REVIEW

**Review by timbot1789 — COMMENTED** [review]
Looks like we may need to update some of the imports in unit tests with the new file paths. I can help with that if you need. Feel free to ping me.

**Comment by timbot1789** [comment]
> As stated in the title, I'm reorganizing utils into 4 general sub-directories:
> 
> * barcode
> * cryptography
> * frontend
> * solid
> 
> Still a bit unsure about using solid as a directory name since it could imply that it's actually a utils folder for Solid rather than PASS, but I don't know of another name at the moment. Still, I'm open to suggestions for all 4 names.
> 
> The file index.js is still being used to funnel all the exports from the sub-directories to the rest of src.
> 
> The full utils directory will look like the following:
> 
> ```
> └── utils
>     ├── barcode
>     │   ├── barcode-date-parser.js
>     │   └── barcode-scan.js
>     ├── cryptography
>     │   ├── credentials-helper.js
>     │   └── credentials-helper.test.js
>     ├── frontend
>     │   ├── FormSubmissionHelper.js
>     │   ├── FormSubmissionHelper.test.js
>     │   ├── form-helper.js
>     │   ├── logout-helper.js
>     │   └── notification-helper.js
>     ├── index.js
>     └── solid
>         ├── session-core.js
>         ├── session-helper.js
>         └── session-helper.test.js
> ```
> 
> When we decide to refactor session-core, we can decide then where to place within these 4 categories or create a new one altogether.

I like the idea of this refactor. It looks like the `solid` folder mostly contains things about networking and sessions. Maybe `network` or `session` are possible names for the folder?

File: src/utils/frontend/FormSubmissionHelper.test.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ describe('FormSubmissionHelper', async () => {
   });
 
   describe('Validations', () => {
-    vi.mock('./session-core', () => ({
+    vi.mock('../solid/session-core', () => ({
```
Comment (timbot1789) [inline src/utils/frontend/FormSubmissionHelper.test.js]: Need to update the path of these module mocks. These mocks basically say "I want the `FormSubmissionHelper` to call session-core, but I don't actually care what session-core does."

Comment (leekahung) [reply src/utils/frontend/FormSubmissionHelper.test.js]: Saw the updates, thanks for going over them with me on the voice chat!

**Comment by leekahung** [comment]
Using 'network' in place of 'solid' sounds like a good idea. Would change to this in a new commit.

File: src/utils/frontend/logout-helper.js
Comment (leekahung) [inline src/utils/frontend/logout-helper.js:1]: Removing removeKeys in favor of localStorage.clear().

**Review by leekahung — COMMENTED** [review]
See comments for request changes.

**Pull request merged**